chore: upgrade docs tooling

ThisIsMissEm opened the pull request:
This upgrades the docs tooling and attempts to clean up the documentation to include only what end-users of the module actually need to know about.

# Checklist

- [ ] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

ThisIsMissEm committed:
chore: remove comments from tsconfig.json to prevent formatting churn

ThisIsMissEm committed:
chore: upgrade documentation tooling

ThisIsMissEm committed:
chore: exclude tests from builds

ThisIsMissEm committed:
chore: add developer script for cleaning up build output

ThisIsMissEm committed:
docs: improve the clarity of the documentation

ThisIsMissEm commented:
@NSeydoux I've added a usage example here: https://solid-client-notifications-js-git-chore-upgrade-d-84d5fc-inrupt.vercel.app/classes/websocketNotification.WebsocketNotification.html

Though it is quite minimal

ThisIsMissEm committed:
docs: support source urls on vercel

NSeydoux commented on docs/api/source/index.rst:
```diff
@@ -1,14 +1,103 @@
+====================================================
 
+`@inrupt/solid-client-notifications
+<https://npmjs.com/package/@inrupt/solid-client-notifications>`__ is a
+JavaScript library for requesting and managing access given to an agent for a
+resource. These access grants are represented by Verifiable Credentials, signed
+by an Issuer associated to the Pod server where the resources are hosted.
```
```suggestion
<https://npmjs.com/package/@inrupt/solid-client-notifications>`__ is a
JavaScript library for subscribing to notifications sent by a Solid server.
These notifications may be sent synchronously e.g., through WebSockets,
or asynchronously e.g. via WebHooks. ? Notifications are typically sent
when a resource is updated/deleted, or when a child resource is added
to/removed from a container.
```

ThisIsMissEm replied:
Have updated it, but with a refined version of the text above.

NSeydoux commented on src/notification.ts:
```diff
@@ -18,36 +18,18 @@
+ * @internal
+ */
 export class BaseNotification {
   topic: string;
```
All of the properties of this class should be private, and not appear in the documentation. All these values are provided by the constructor or during the negotiation with the notification endpoint, they should neither be set a posteriori not be read by a third party.

ThisIsMissEm committed:
chore: rename webSocket property and mark as internal

ThisIsMissEm committed:
chore: rename BaseNotificationOptions to NotificationOptions

ThisIsMissEm committed:
chore: upgrade browser testApp

This updates the dependencies, build target, and uses `file:` referenced version of this SDK when testing, no more `npm link`

ThisIsMissEm commented on e2e/browser/testApp/package.json:
```diff
@@ -10,17 +10,17 @@
     "@inrupt/solid-client-authn-browser": "^1.11.7",
-    "@inrupt/solid-client-notifications": "*",
-    "next": "12.1.2",
+    "@inrupt/solid-client-notifications": "file:../../../",
```
@NSeydoux I switched this to use `file:` instead, after the discussion with the npm team here: https://github.com/npm/feedback/discussions/667
